util/tracing/detect/detect.go

```diff
@@ -90,6 +90,10 @@ func detect() error {
 		return nil
 	}
 
+	exp = &threadSafeExporterWrapper{
```

Comment:
This needs to be before the `Recorder` switch above in line 84. If no exporter is configured but `Recorder` is configured then `exp` will be `Recorder`. If both are configured then `exp` is `Recorder` that wraps the `threadSafeExporterWrapper`

Reply:
Right, sorry about that. I missed the changes there (unfortunately no merge conflict :p)